src/lib.rs

```diff
@@ -8,35 +8,246 @@ mod transaction;
+
+pub struct Channel {
+    x_self: OwnershipKeyPair,
+    X_other: OwnershipPublicKey,
+    pub TX_f_body: FundingTransaction,
+    pub current_state: ChannelState,
+    revoked_states: Vec<RevokedState>,
+}
+
+#[async_trait::async_trait]
+pub trait BroadcastSignedTransaction {
+    async fn broadcast_signed_transaction(&self, transaction: Transaction) -> anyhow::Result<()>;
+}
+
+#[async_trait::async_trait]
+pub trait SendMessage {
+    async fn send_message(&mut self, message: Message) -> anyhow::Result<()>;
+}
+
+#[async_trait::async_trait]
+pub trait ReceiveMessage {
+    async fn receive_message(&mut self) -> anyhow::Result<Message>;
+}
+
+impl Channel {
+    /// Create a channel in the role of Alice.
+    ///
+    /// The `fund_amount` represents how much Bitcoin Alice will contribute to
+    /// the channel. Bob will contribute the _same_ amount as Alice.
+    ///
+    /// Consumers should implement the traits `SendMessage` and `ReceiveMessage`
+    /// on the `transport` they provide, allowing Alice to communicate with
+    /// Bob.
+    pub async fn create_alice<T, W>(
+        transport: &mut T,
+        wallet: &W,
+        fund_amount: Amount,
+        time_lock: u32,
+    ) -> anyhow::Result<Self>
+    where
+        W: BuildFundingPSBT + SignFundingPSBT + BroadcastSignedTransaction,
+        T: SendMessage + ReceiveMessage,
+    {
+        let alice0 = create::Alice0::new(fund_amount, time_lock);
+
+        let message0_alice = alice0.next_message();
+        transport
+            .send_message(Message::CreateMessage0(message0_alice))
+            .await?;
+
+        let message0_bob = match transport.receive_message().await? {
+            Message::CreateMessage0(message) => message,
+            _ => anyhow::bail!("wrong message"),
+        };
+        let alice1 = alice0.receive(message0_bob, wallet).await.unwrap();
+
+        let message1_alice = alice1.next_message();
+        transport
+            .send_message(Message::CreateMessage1(message1_alice))
+            .await?;
+
+        let message1_bob = match transport.receive_message().await? {
+            Message::CreateMessage1(message) => message,
+            _ => anyhow::bail!("wrong message"),
+        };
+        let alice2 = alice1.receive(message1_bob).unwrap();
+
+        let message2_alice = alice2.next_message();
+        transport
+            .send_message(Message::CreateMessage2(message2_alice))
+            .await?;
+
+        let message2_bob = match transport.receive_message().await? {
+            Message::CreateMessage2(message) => message,
+            _ => anyhow::bail!("wrong message"),
+        };
+        let alice3 = alice2.receive(message2_bob).unwrap();
+
+        Self::create(transport, wallet, alice3).await
+    }
+
+    /// Create a channel in the role of Bob.
+    ///
+    /// The `fund_amount` represents how much Bitcoin Bob will contribute to
+    /// the channel. Alice will contribute the _same_ amount as Bob.
```

Comment:
```suggestion
    /// the channel. Alice must contribute the _same_ amount as Bob.
```
Also same time lock?